polus-recycle-vector-plugin/src/filepattern.py

```diff
@@ -0,0 +1,579 @@
+import re, copy
```

Comment:
Whoo, I just started reviewing this AGAIN. Hah got me. I was reading it and was like, boy this is more complicated than I thought... for the third time that I've read the exact same code, hah. Please update with file pattern utility.